smithery-mcp-server/src/services/document-ingestion.service.ts

```
@@ -0,0 +1,362 @@
+    try {
+        console.log(`[DocDiscovery] Attempt 3: Mapping site with Firecrawl 'crawl' (shallow) for ${baseUrl}`);
+        const toolboxResponse = await ctx.callTool(
+            '@smithery/toolbox',
+            'use_tool',
+            {
+                target_server_id: 'firecrawl-mcp-server',
+                tool_name: 'crawl',
+                tool_args: { url: baseUrl, maxDepth: 1, limit: 100, scrapeOptions: { formats: ['links'] } } // Get links only
+            }
+        );
+        if (toolboxResponse && toolboxResponse.content && toolboxResponse.content[0] && toolboxResponse.content[0].type === 'application/json') {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)